### PR TITLE
fix: Don't show join default channels option for edit user form 

### PR DESCRIPTION
**Pull request by yash-rajpal**
## Proposed changes (including videos or screenshots)

When we create new users, we have to option to add these new users to default channels.
We were also showing this option on edit users panel, which doesn't make any sense, so this PR removes this option.
## Further comments

SUP-423

### CODE REVIEW

**Review by Gustrb — APPROVED** [review]

File: .changeset/real-bobcats-train.md [hunk 1]
```diff
@@ -0,0 +1,6 @@
+---
+'@rocket.chat/rest-typings': patch
+'@rocket.chat/meteor': patch
+---
+
+Don't show Join default channels option on edit user form and stop accepting this param on API
```
Comment (MarcosSpessatto) [inline .changeset/real-bobcats-train.md:6]: We can't just remove the parameter right now in a sudden manner, it's considered a breaking change.

File: packages/rest-typings/src/v1/users/UsersUpdateParamsPOST.ts [hunk 2]
```diff
@@ -16,7 +16,6 @@ export type UsersUpdateParamsPOST = {
 		nickname?: string;
 		statusText?: string;
 		roles?: string[];
-		joinDefaultChannels?: boolean;
```
Comment (MarcosSpessatto) [inline packages/rest-typings/src/v1/users/UsersUpdateParamsPOST.ts:19]: We can't just remove the parameter right now in a sudden manner, it's considered a breaking change. You can still keep the client-side changes though.

**Review by MarcosSpessatto — CHANGES_REQUESTED** [review]

File: packages/rest-typings/src/v1/users/UsersUpdateParamsPOST.ts (hunk 2 above)
Comment (yash-rajpal) [reply packages/rest-typings/src/v1/users/UsersUpdateParamsPOST.ts:19]: You are right, I'll revert this.

**Review by MarcosSpessatto — CHANGES_REQUESTED** [review]
I think we could add a UI test to cover this behavior. WDYT?

**Review by MarcosSpessatto — APPROVED** [review]

**Pull request merged**